widgetschema.json

```diff
@@ -0,0 +1,244 @@
+                {
+                    "properties": { "type": { "const": "inputGroup" } },
+                    "required": ["name", "type", "inputGroup"]
+                },
+                { "properties": { "type": { "const": "actionButtons" } } },
+                { "properties": { "type": { "const": "documentUpload" } } },
+                { "properties": { "type": { "const": "columnSelector" } } },
+                { "properties": { "type": { "const": "filter" } } },
+                {
+                    "properties": { "type": { "const": "text" } }
+                },
+                { "properties": { "type": { "const": "number" } } },
+                { "properties": { "type": { "const": "date" } } },
+                { "properties": { "type": { "const": "boolean" } } },
+                {
+                    "properties": { "type": { "const": "choices" } },
+                    "oneOf": [{ "required": ["name", "type", "choices"] }, { "required": ["name", "type", "api"] }]
+                }
+            ],
+            "properties": {
+                "name": {
+                    "type": "string",
+                    "description": "This is the name of the input field, and also the property name passed to the widget react component"
+                },
+                "type": {
+                    "type": "string",
+                    "description": "This is the “type” of the input and will determine which input field(s) are presented to the builder user, and the format of the values passed to the widget.",
+                    "enum": [
+                        "text",
+                        "number",
+                        "boolean",
+                        "choices",
+                        "inputGroup",
+                        "date",
+                        "actionButtons",
+                        "documentUpload",
+                        "columnSelector",
+                        "filter"
+                    ]
+                },
+                "displayName": {
+                    "type": "string",
+                    "description": "The text that shows up in the list of widget-settings while editing a widget. If “displayName” is not specified, the value specified for “name” is displayed in start-case."
+                },
+                "instructionText": {
+                    "type": "string",
+                    "description": "This value shows up as a prompt at the top of all the input field(s) for that setting."
+                },
+                "inputLabel": {
+                    "type": "string",
+                    "description": "If applicable, this value shows up near/next to the input field."
+                },
+                "isOptional": {
+                    "oneOf": [{ "$ref": "#/definitions/widgetSettingOptionalValue" }, { "type": "boolean" }],
+                    "description": "Indicates if the property is optional"
+                },
+                "isWidgetTitle": {
```

Comment:
This is being using in the Page Header and List Table

Reply:
But what does it mean to be flagged as "widget title"?  @bradwray

Reply:
@actong this appears to be what is used to add a title to the widgets in builder in the grid layout so that mustard would be able to visually identify the right widget to click. It was added with CDR-1128.